### ISSUE TITLE
Adding souls upon taking armor does not run for every armor piece when activated

### ISSUE THREAD

**Issue by ArmonicPvP** [GC-spigot/AdvancedEnchantments#4238]
### Describe the bug

When adding souls upon taking damage, the effect only runs for one piece, not all of them.

### How to reproduce

soulforged:
  display: '%group-color%Soulforged'
  description: Chance to collect souls when attacked
  applies-to: Armor
  type: DEFENSE;DEFENSE_MOB;DEFENSE_MOB_PROJECTILE;DEFENSE_PROJECTILE
  group: SOUL
  applies:
    - ALL_ARMOR
  levels:
    '1':
      chance: 100
      cooldown: 3
      effects:
        - ADD_SOULS:1 @Victim
    '2':
      chance: 100
      cooldown: 2
      effects:
        - ADD_SOULS:<random number>1-2</random number> @Victim
    '3':
      chance: 100
      cooldown: 1
      effects:
        - ADD_SOULS:<random number>1-3</random number> @Victim

### "/ae plinfo" link

https://paste.md-5.net/imetozurel

**Comment by GC-spigot** [comment]
Why would this add to all armor pieces? It only adds to the item which armor is worn on.

**Comment by ArmonicPvP** [comment]
Let me rephrase: it appears to only run once ehen successful when it should check for each armor piece. So if you have it on all 4 pieces, it will run once for each piece. Instead, it seems to only run once total, even if each piece of armor activates the ability on its own piece.
